lib/beaker/ssh_connection.rb

```diff
@@ -147,12 +149,16 @@ def wait_for_connection_failure options = {}, stdout_callback = nil, stderr_call
         slept = 0
         stdout_callback.call("sleep #{wait} second(s): ")
         while slept < wait
-          sleep slept
+          sleep 1
           stdout_callback.call('.')
           slept += 1
         end
         stdout_callback.call("\n")
-        (last_wait, wait) = wait, last_wait + wait
+        if poll_interval
+          wait = poll_interval
+        else
+          (last_wait, wait) = wait, last_wait + wait
+        end
```

Comment:
@samwoods1 I'm down with the motivation here, and having the toggle-ability of using the pattern that fits you, but I still don't think the logic should be executed here.

Considering what you need is the ability to toggle between the fibonacci backoff or a static poll timer, a method should be added to `repetition.rb` that allows for what you're looking for, pulling the mechanics of repetition out of `wait_for_connection_failure`. That method would look something like this:
```ruby
def repeat_choosing_method_for(max_tries=11, poll_interval=nil, &block)
  if poll_interval
    repeat_for(poll_interval, block)
  else
    repeat_fibonacci_style_for(max_attempts, block)
  end
end
```

Once that's implemented, all the repetition logic will be out of the `wait_for_connection_failure` method, and it won't have to care about any of that. This has a few benefits over the current implementation:

1. you can use this toggle-ability between repetition-types in other places in your testing
1. it's less likely that there will be timing bugs like this, the more centralized the repetition implementations are